lib/valueflow.cpp

```diff
@@ -4259,26 +4259,30 @@ struct LifetimeStore {
     }
 };
 
-static bool isOwningVariables(const std::list<Variable>& vars, int depth = 10)
+static bool hasBorrowingVariables(const std::list<Variable>& vars, const std::vector<const Token*>& args, int depth = 10)
 {
     if (depth < 0)
         return false;
-    return vars.empty() || std::all_of(vars.cbegin(), vars.cend(), [&](const Variable& var) {
-        if (var.isReference() || var.isPointer())
-            return false;
+    return std::any_of(vars.cbegin(), vars.cend(), [&](const Variable& var) {
         const ValueType* vt = var.valueType();
         if (vt) {
-            if (vt->pointer > 0)
+            if (std::none_of(args.begin(), args.end(), [vt](const Token* arg) {
```

Comment:
This should be `if (vt->pointer > 0  && std::none_of(args.begin(), args.end(), ...))`. I haven't tested on this PR, but it looks like without this check it could lead to FPs in cases like this:

```cpp
struct A {
    explicit A(int& i) {
        data = std::span<int>{&i, 1};
    }
    void foo() const {
        for(auto& i:data) i++;
    }
    std::span<int> data;
};

int f() {
    int i = -1;
    A a{i};
    i++;
    a.foo();
    return 1 / i;
}
```

Reply:
There already is a FP for this, and `if (vt->pointer > 0 ...` does not prevent it.